### ISSUE TITLE
chatbot UI failed to load Error: postMessage from invalid origin https://code.gist.build

### ISSUE THREAD

**Issue by FritsDonk** [aws-samples/aws-lex-web-ui#730]
Hello,

I deployed the stack in London (Lex Web UI CloudFormation template (v0.21.5))
The `parent.html` example application works fine on the stack.

I copied the "Lex Web UI Iframe Snippet" from `iframe-snippet.html` into my website and configured the `iframeOrigin` and `parentOrigin` parameters in both the JSON config file and the js code.

![image](https://github.com/user-attachments/assets/b339c56d-bcf0-4504-b802-6de084fc3da1)

I get the following error in the browser developer tools console window:

![image](https://github.com/user-attachments/assets/18f9a88d-0dec-4b8f-9e3a-02d667e06837)

I tried the standalone installation too, in a sub-folder on the site and it seemed to not carry the baseUrl through to all the dependencies, so I published it to it's own site (subdomain) in the root folder and again the `parent.html` example application worked fine, but when I tried to use that in my iFrame I have similar issues with origin.

It seems like the parentOrigin parameter is not working at all.

I don't really want to publish the standalone dist folder content to the root of my site if it can be avoided, so I am stuck.

Please can you assist?

**Comment by atjohns** [comment]
When you deploy the CFN template what are you using for `WebAppParentOrigin` and `WebAppPath`? 

I don't think you should need to send anything in `chatbotUiConfig` if those parameters are set correctly in the deployment.

**Comment by FritsDonk** [comment]
Thanks for getting back to me. I did not fill those in during deployment.
Currently they are the default values.
WebAppParentOrigin = ""
WebAppPath = "/parent.html"

The description seems to suggest that is where the sample parent page will be hosted, and I wanted to leave that in S3

![image](https://github.com/user-attachments/assets/9a4c4168-f618-4fb8-96ae-3b3fff769e6f)

I have now updated that parameter in CF and it seems to have solved the problem and it started working. The warning `postMessage from invalid origin https://code.gist.build` still shows, but everything is working.

But I have decided that I want to run this locally within my site, not using Cloud Formation. And I have managed to get all that to work fine, when I copied all the dist assets into the root of my site. I would only ask if it is possible to have all the assets in a sub-folder, as per my other ticket https://github.com/aws-samples/aws-lex-web-ui/issues/728
